csm_web/frontend/src/components/MentorSection.js

```diff
@@ -247,7 +259,132 @@ class MentorSectionAttendance extends React.Component {
   }
 }
 
+function zeroPadTwoDigit(num) {
+  return num < 10 ? `0${num}` : num;
+}
+
+class SpacetimeEditModal extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { location: "", day: "Mon", time: "", isPermanent: false, changeDate: "", showSaveSpinner: false };
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  static propTypes = { closeModal: PropTypes.func.isRequired, spacetimeId: PropTypes.number.isRequired };
+
+  handleChange({ target: { name, value } }) {
+    this.setState({ [name]: value });
+  }
+
+  handleSubmit(event) {
+    event.preventDefault();
+    const { closeModal, spacetimeId } = this.props;
+    let { location, day, time, isPermanent, changeDate } = this.state;
+    isPermanent = !!isPermanent;
```

Comment:
in what case would isPermanent be non-boolean?

Reply:
Always, in fact. The `handleChange` function receives `value` from the target element and uses it to update state, and `value` is always treated as a string.